cmd/capibmadm/cmd/powervs/powervs.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package powervs
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	networkcmd "sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/cmd/powervs/network"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/util"
+)
+
+// NewPowerVSCommand initialises and returns powervs command.
+func NewPowerVSCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "powervs",
+		Short: "Commands for operations on PowerVS resources",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			apiKey, err := util.GetEnv(options.IBMCloudAPIKeyEnvName)
+			if err != nil {
+				return fmt.Errorf("ibmcloud api key is not provided, set %s environmental variable", options.IBMCloudAPIKeyEnvName)
+			}
```

Comment:
lets move this check to root command because we need API key for both vpc and powervs

Reply:
There is a nice [workaround](https://github.com/spf13/cobra/issues/252#issuecomment-622561414) mentioned here, wondering if we can adopt that because we might end up writing some more checks in PersistentPreRunEs in multiple places.